utils_cv/detection/plot.py

```diff
@@ -116,6 +131,98 @@ def display_bounding_boxes(
         plt.show()
 
 
+def plot_grid(
+    plot_func: Callable[..., None],
+    args: Union[Callable, Iterator, Any],
+    rows: int = 1,
+    cols: int = 3,
+    figsize: Tuple[int, int] = (16, 16),
+) -> None:
+    """ Helper function to plot image grids.
+
+    Args:
+        plot_func: callback to call on each subplot. It should take an 'ax' as
+        the last param.
+        args: args can be passed in in many forms. It can be an iterator, a
+        callable, or simply some static parameters. If it is an iterator, this
+        function will call `next` on it each time. If it is a callable, this
+        function will call the function and use the returned values each time.
+        rows: rows to plot
+        cols: cols to plot, default is 3. NOTE: use cols=3 for best looking
+        grid
+        figsize: figure size (will be dynamically modified in the code
+
+    Returns nothing but plots graph
+    """
+    fig_height = rows * 8
+    figsize = (16, fig_height)
+
+    fig, axes = plt.subplots(rows, cols, figsize=figsize)
+
+    if rows == 1:
+        axes = [axes]
+
+    for row in axes:
+        for ax in row:
+            # dynamic injection of params into callable
+            arguments = (
+                args()
+                if isinstance(args, Callable)
+                else (next(args) if hasattr(args, "__iter__") else args)
+            )
+            try:
+                plot_func(arguments, ax)
+            except Exception:
+                plot_func(*arguments, ax)
+
+    plt.subplots_adjust(top=0.8, bottom=0.2, hspace=0.1, wspace=0.2)
+
+
+def plot_detection_vs_ground_truth(
+    im_path: str,
+    detection_annos: List[AnnotationBbox],
```

Comment:
Not sure those two variable names are consistent with your other code: "detection_annos" and "ground_truth_annos". Maybe rename to annno_bboxes and pred_bboxes but not sure right now this is how to call them.

Reply:
mm i see.. i think i've been thinking of the word 'annotation' a bit differently; like a prediction can produce a set of annotations. But i think you're right about this being confusing... the word "annotation" should strictly mean labelled data, in which case we'll need another word to represented predicted data. I'll go through the code base and make this update. Sorry for this confusion.

Reply:
If it makes sense, i'll implement it so that DetectionBbox inherits from AnnotationBbox which inherits from Bbox. 

I think the more logically intuitive implementation is that both DetectionBbox and AnnotationBbox inherit from Bbox, but practically, the DetectionBbox shoudl contain everything that the AnnotationBbox contains. I'll also make Bbox a 'private' class, so to speak, prefixing it's name with and underscore: `_Bbox`. This is to indicate that only the DetectionBbox or AnnotationBbox be used.